fdbserver/OldTLogServer_6_2.actor.cpp

```diff
@@ -439,17 +441,34 @@ struct LogData : NonCopyable, public ReferenceCounted<LogData> {
 					} else {
 						sizes.second -= m.second.expectedSize();
 					}
+					sizeErasedTotal += m.second.expectedSize();
 
 					self->versionMessages.pop_front();
 				}
 
+				messageErasedTotal += messagesErased;
 				int64_t bytesErased = messagesErased * SERVER_KNOBS->VERSION_MESSAGES_ENTRY_BYTES_WITH_OVERHEAD;
 				logData->bytesDurable += bytesErased;
 				tlogData->bytesDurable += bytesErased;
 				tlogData->overheadBytesDurable += bytesErased;
 				wait(yield(taskID));
 			}
 
+			int64_t lag = logData->durableKnownCommittedVersion - before + 1;
+			if (taskID == TaskPriority::TLogPop && //trace only when eraseMessagesBefore is called by TLogPop
+				(!self->nothingPersistent || messageErasedTotal != 0) && //see any data removed on disk or in memory
+				(self->unpoppedRecovered || lag >= SERVER_KNOBS->TLOG_LONG_LAG_THRESHOLD)) { //when recovery or long lag
+				TraceEvent("EraseMessages", logData->logId)
```

Comment:
As you mentioned, this message could be generated too many. Can you do an analysis on the frequency of this message and see if it is reasonable?

We can also add a knob to control whether to print this message or not, just in case this ends up being too many in prod, and we want to disable it.

Reply:
This message only appears when recovery happens or when the lag is larger than 2/5 * STORAGE_DURABILITY_LAG_SOFT_MAX.